File: PartnerApi/Ruby/b2_list_groups.rb
# frozen_string_literal: true

require 'net/http'
require 'uri'
require 'json'

api_url = '' # Provided by b2_authorize_account
auth_token = '' # Provided by b2_authorize_account
admin_account_id = '' # Provided by b2_authorize_account

uri = URI("#{api_url}/b2api/v4/b2_list_groups")

request = Net::HTTP::Post.new(uri)
request['Authorization'] = auth_token
request['Content-Type']  = 'application/json'
request.body = { adminAccountId: admin_account_id }.to_json

response = Net::HTTP.start(uri.host, uri.port, use_ssl: uri.scheme == 'https') do |http|
  http.request(request)
end

puts response.body

File: PartnerApi/Python/b2_list_groups.py
import json
import requests

api_url = ""  # Provided by b2_authorize_account
account_authorization_token = ""  # Provided by b2_authorize_account
admin_account_id = ""  # Provided by b2_authorize_account

response = requests.post(
    f"{api_url}/b2api/v4/b2_list_groups",
    headers={"Authorization": account_authorization_token},
    json={"adminAccountId": admin_account_id},
)
response_data = response.json()
print(json.dumps(response_data, indent=2))
